packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx

```diff
@@ -190,7 +190,8 @@ export const EditView = () => {
         const res = await createToken({
           ...body,
           // lifespan must be "null" for unlimited (0 would mean instantly expired and isn't accepted)
-          lifespan: body?.lifespan || null,
+          lifespan:
+            body?.lifespan && body.lifespan !== '0' ? parseInt(body.lifespan.toString(), 10) : null,
```

Comment:
why are you converting it to a string to then parse it back to a number? Is it not already a number prior?

Reply:
You can find the reasoning [here](https://github.com/strapi/strapi/pull/19621)

Reply:
that is so confusing

Reply:
I just added one